Modules/_threadmodule.c

```diff
@@ -890,25 +877,20 @@ lock_repr(PyObject *op)
 {
     lockobject *self = (lockobject*)op;
     return PyUnicode_FromFormat("<%s %s object at %p>",
-        self->locked ? "locked" : "unlocked", Py_TYPE(self)->tp_name, self);
+        PyMutex_IsLocked(&self->lock) ? "locked" : "unlocked", Py_TYPE(self)->tp_name, self);
 }
 
 #ifdef HAVE_FORK
 static PyObject *
 lock__at_fork_reinit(PyObject *op, PyObject *Py_UNUSED(args))
 {
-    lockobject *self = (lockobject*)op;
-    if (_PyThread_at_fork_reinit(&self->lock_lock) < 0) {
-        PyErr_SetString(ThreadError, "failed to reinitialize lock at fork");
-        return NULL;
-    }
-
-    self->locked = 0;
-
+    lockobject *self = (lockobject *)op;
+    self->lock = (PyMutex){0};
```

Comment:
This works fine, but you can also use `_PyMutex_at_fork_reinit(&self->lock)`

Reply:
fixed